heron/spi/src/java/com/twitter/heron/spi/statemgr/IStateManager.java

```diff
@@ -51,6 +51,64 @@
  */
 
 public interface IStateManager extends AutoCloseable {
+  enum StateLocation {
+    TMASTER_LOCATION("tmasters", "TMaster location"),
+    TOPOLOGY("topologies", "Topologies"),
+    PACKING_PLAN("packingplans", "Packing plan"),
+    PHYSICAL_PLAN("pplans", "Physical plan"),
+    EXECUTION_STATE("executionstate", "Execution state"),
+    SCHEDULER_LOCATION("schedulers", "Scheduler location");
+
+    private final String dir;
+    private final String name;
+
+    StateLocation(String dir, String name) {
+      this.dir = dir;
+      this.name = name;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public String getDirectory(String root) {
+      return concatPath(root, dir);
+    }
+
+    public String getNodePath(String root, String topology) {
+      return concatPath(getDirectory(root), topology);
+    }
+
+    public void delete(IStateManager stateManager, String topology) {
+      switch(this) {
+        case TMASTER_LOCATION:
+          stateManager.deleteTMasterLocation(topology);
+          break;
+        case TOPOLOGY:
+          stateManager.deleteTopology(topology);
+          break;
+        case PACKING_PLAN:
+          stateManager.deletePackingPlan(topology);
+          break;
+        case PHYSICAL_PLAN:
+          stateManager.deletePhysicalPlan(topology);
+          break;
+        case EXECUTION_STATE:
+          stateManager.deleteExecutionState(topology);
+          break;
+        case SCHEDULER_LOCATION:
+          stateManager.deleteSchedulerLocation(topology);
+          break;
+        default:
+          // This should never be reached, nothing to do.
```

Comment:
This will be reached if someone adds a new Enum without updating the case statement. Please throw an `IllegalArguementException` (?) here. Bonus points for a way to catch such a condition at compile time, which I can't think of atm.